### DISCUSSION TITLE
Community types

### DISCUSSION THREAD

**Discussion by megkrawchuk** [HaireLab/community-stress-gradients#6] [General]
Hi everyone.
I've completed a first round of allocating each of our field plots to a community type using the PAM clustering algorithm, and vegetation cover from the growth form/phenology trait groupings I described earlier.

I'll add more information soon, but just wanted to share that I'm moving along. These community types aim to be the anchor of our analyses, paired with the climate/stress metrics Sandra is working on.
Wow, does this mean we're getting close to a first formal analysis/exploration?! Exciting.

PAM suggests we do well with five community types.  Seems to be more robust/effective than k-means for our data (some outlier issues, e.g., mischievous clusters with n=1 in k means). I need to explore them a bit more, to characterize dominant community members and their traits. More soon. But here's the clustering, as a teaser:
![pam optimal cluster try5](https://user-images.githubusercontent.com/84547345/124042166-218f0700-d9bd-11eb-91c4-16ba2fc342f7.jpeg)

**Comment by megkrawchuk** [comment]
Here's the link to the process I used, described in another Discussion thread

https://github.com/HaireLab/community-stress-gradients/discussions/1#discussioncomment-765953

**Reply by HaireLab** [reply]
I like this approach and the outputs thus far. Should be an efficient way to filter the data for the second round. My only question is, what output does the clustering algorithm give you that allows you to rank/id  dominant community members? I haven't done clustering for a while...

**Comment by megkrawchuk** [comment]
**COMMUNITY TYPES. Update 09/07/2021.**
I've re-worked the community types analyses. See below, ignore the reporting above (same general approach, just different clustering algorithm). The goal of this stage of the work was to assign each plot (here focused on the non-refugia plots – early seral, “OUT” plots) to a group based on some basic functional metrics. This approach allows us to include all the SW and NW sites from the 13 fires without getting concerned about different species across the wide environmental gradient. Instead, we focus on functional groupings. 

The goal is to use clustering algorithms to assign like with like, based on the composition of functional traits from all species in each plot.Initial scoping of available functional traits for species in our samples suggested many gaps in the more nuanced functional traits. So here I focus on developing attributes for each species based on growth form and phenology, as the basis for our groupings.

Using the All_Refugia_Plants excel file, I worked with the “cover” tabulation and the “list” tabulation.

**Step 1**: Using All_Refugia_Plants$list to finalize growth form and phenology attributes, and named that community types.

Each species was assigned a growth form (growthform: forb, gram, tree, shru, subs) based on GROWTH HABIT (column J) and HABIT2/HABIT3. There were a few nuanced decisions, but overall quite straightforward. Additional nuances are recorded in my own notes but not recorded here. I will need to get confirmation for where Will and Jonathan obtained the growth form data they used in the original spreadsheet.

Each species was assigned phenological attributes (timing: annual, perennial, deciduous, evergreen), crosswalked with information in original dataset and from research online to fill in any gaps. If annual/biennial then kept as annual. Additional nuances are recorded in my own notes but not recorded here.

The growth form and phenological attributes were combined to develop functional groups to which every species is assigned. 
Originally, I included a subshrub grouping, but after some preliminary analysis decided to combine subshrubs with shrubs because of low cover values overall in the subshrub categories.

Final eight groups for original assignment: forb.annual, forb.perennial, graminoid.annual, graminoid.perennial, shrub.deciduous, shrub.evergreen, tree.deciduous, tree.evergreen, 

**Step 2:** Merge and organize data so that each plot entry has attributes for the eight newly developed functional groups. These attributes are the cover estimates for any species with heights <=5m. This coincides with strata 2-5 (excluding stratum 1 which is >5m). This matches Will’s criterion in community composition paper, where he used cover estimates to <=5m.

One plot was removed, PUMPOUT005, based on communication with Sandra/Jonathan that this plot had incorrect location data.

**Step 3:** Clustering of plots based on community composition based on cover of the eight functional groups. The outcome will be a set of COMMUNITY TYPES that will be included in statistical analyses. Another way to do this is to calculate the dominant functional group for each plot, but this ignores the overall community composition.

I used agglomerative clustering for this step. From what I’ve researched, the agglomerative clustering (cluster library) is more stable than PAM and far more robust than k-means. I used Ward’s methods, which is apparently preferable when you’re concerned with uneven cluster sizes. I compared four different methods (average, single, complete, ward) and Ward’s was identified as the strongest analytically.

I used the agnes function on scaled functional group data. The dendrogram and outputs suggested cutting the tree at six types. However the number of plots identified for each group is somewhat unbalanced. This imbalance, and particularly the “impenetrable” group 1 shows up using a variety of different techniques (I can share details as needed). 
 1       2    3    4    5    6 
342  14  39  57  51  39 

Dendrogram (If you zoom in you can see the six clusters identified by different colored boxes): 

![image](https://user-images.githubusercontent.com/84547345/132428892-afc562ec-57cc-4fcf-86f3-b568465d4ebe.png) 

Clusters presented in NMDS ordination space:

![image](https://user-images.githubusercontent.com/84547345/132428941-41deb671-bd80-43f3-974a-7c4c633b8796.png)
I did a simple assessment to characterize the differences among clusters, based on the eight functional groups:

 
![image](https://user-images.githubusercontent.com/84547345/132428973-da77dae5-ec21-4aac-99af-ca31dfa9a33b.png)

cluster1 (342): no clear dominance by any particular type; catch all
cluster2 (14): higher values of gram.a, and forb.a 
cluster3 (39): higher values of shru.e1, moderate gram.p
cluster4 (57): higher values of shru.d1, moderate gram.p
cluster5 (51): highest values of tree.e, some higher gram.p, some higher tree.d
cluster6 (39): highest value of tree.d

I then opted to take a hierarchical approach and do a second round of clustering on group 1 alone to further divide up those 342 plots into finer categories. Cluster 1 shows no clear dominance by any particular functional group, it’s sort of a catch-all – so divvying up this cluster could be tricky.

Note, cluster 2 only has 14 plots assigned to it – it’s the annual grasses and forbs bin. We might find that this small membership (and unbalanced design) constrains some of our further statistical analyses – so let’s keep an eye on it.

**Step 4:** Second level of agglomerative clustering, focused on cluster 1 only. This analysis suggested either 2 or 3 clusters were appropriate. I decided to go with two clusters, which would bring the final COMMUNITY TYPES to eight. Further divisions will be difficult to interpret and defend. Though maybe further exploration of dominant vegetation and additional functional trait could help break this us further, or differently. 

clust
  1   2 
158 184

Dendrogram: You can see the two clusters identified in this second analyses.

 ![image](https://user-images.githubusercontent.com/84547345/132435443-fc53895c-46a8-48f9-a939-b6cc974acf45.png)

And boxplots by functional groups shows that cluster 1 has higher cover of perennial graminoids, sometimes higher perennial forbs. So perennial composition of this generally herbaceous plot type makes this cluster slightly unique.
Cluster sometimes has higher deciduous tree cover, lower perennial graminoid cover, and lower perennial forb. So more deciduous trees and lower perennial herbaceous species in comparison to others.

![image](https://user-images.githubusercontent.com/84547345/132435486-a88b7b10-de7b-4184-8a28-57465e8637ee.png)

**Step 5**: Weaving the clusters together into the dataset to establish the final seven COMMUNITY TYPES.

Here’s the final tally, with TYPES 1 and 7 having the highest membership:
  1      2   3     4   5     6    7 
158  14  39  57  51  39 184

Boxplots show the general distinctions of the seven TYPES:
 
![image](https://user-images.githubusercontent.com/84547345/132435527-c1678f07-1b4b-49b1-b0ae-b2d5f0d73b18.png)

Here’s the general characterization, with diagnostic features of each COMMUNITY TYPE:

type 1. A bit of everything, but higher perennial graminoids and sometimes higher perennial forbs
type 7. A bit of everything, but higher deciduous trees, evergreen trees, evergreen shrubs than type 1 (type 1 and type 7 were the outcomes of the second round of clustering), lower amoungs of perennial graminoids and forbs.
type 2. Higher values of annual fobs and annual graminoids
type 3. Higher values of evergreen shrubs, moderate levels of perennial graminoids
type 4. Higher values of deciduous shrubs, moderate levels of perennial graminoids
type 5. Higher values of evergreen trees, some higher deciduous trees, perennial graminoids
type 6. Higher values deciduous trees

And if we cross walk these against the dominant functional group for each plot we see the agglomerative clustering is generally echoing the dominant composition:

             forb.a forb.p1 gram.a gram.p shru.d1 shru.e1 tree.d tree.e
  1          3          39        2            93         13         8          0       0
  2         2           0          9             0           3          0          0       0
  3         0           0          0            5            0        34          0       0
  4         0           2          0            6          49         0           0      0
  5         0           3          0            5            2          1           1     39
  6         0           3          0            1            1           0         34      0
  7         0          17          0           62         39        37        23      6

Now, onto exploratory analyses. With the environmental data but also to check out dominant species within the types.

**Reply by HaireLab** [reply]
Really nice summary, Meg-- and this documentation will be very useful going forward.
It seems there is usually one group with low membership,  based on past experience. I agree, good to see what this one tells us.

**Reply by cmtortorelli** [reply]
Great to see these coming together! Thanks for this summary, Meg

For your growth form data Q: I extracted these from the USDA plants database for all of Will's species

The clustering seems like a nice approach for continuing on, but if we do end up running into issues because of the low group membership, we could consider some sort of continuous plot metric (e.g. cover of annual grams) as an alternative. Are we considering any other traits (e.g. regeneration strategy - seeders vs sprouters) for future groupings?

**Comment by megkrawchuk** [comment]
You thought you'd never see the day... but here we go! @HaireLab @cmtortorelli 

### **Community matters for Ponderosa Pine recruitment after high-severity fire -- 09/19/2022 update**
Finally, time for an update on community matters! 
Otherwise known as an investigation into the potential influence of the broader functional plant community on post-fire regeneration of ponderosa pine in patches of stand replacing from high severity fire effects. 

Thanks to a pointer from Sandra, I’ve been using [Gavin Simpson’s framework of hierarchical generalized models (with mgcv)](https://peerj.com/articles/6876/). This analytical framework allows us to test for differences in response shapes and forms among groups. In this case, our groups are the community types I developed last year that assign a community type to each plot of our full dataset of 13 fires. In that analysis, I developed seven community types using growth form and phenology, which allowed us to think about functional types across the broad biophysical gradient of samples sites and plant species from OR, ID, CO, NM, AR.  

This is our first analysis that leverages all 13 of the fires we have data on. And the niche this analysis fills is an opportunity to think about “whether community matters?” to post-fire PIPO recruitment. This complements all the existing analyses focused on role of seed source (refugia) and climate on post-fire regen in western forests.

**Hypothesis: Community types will exhibit different relationships for PIPO recruitment because of the functional interactions and ecologies within those communities. Specifically, the shapes of these relationships will differ across a water balance gradient, following the stress gradient hypothesis.** 

So we’re looking for a couple of things. After account for distance to seed source (here using distance to PIPO refugium measured in the field, pinpon.dist.600) we’re interested in: 
a)	are there different functional forms among community types when assessing them across a water balance-relevant explanatory variable
b)	are the relationships interpretable across the water balance-relevant gradient in the context of what we know of the stress gradient hypothesis.

In our brainstorming a while back we developed a bunch of more detailed hypotheses for how and why relationships might differ among community types. We can revisit that later. First, let’s just dig into the analyses to see where we’re at.

One thing we need to keep in mind is that even though I’m testing ideas about whether the broader plant community might play a role in shaping post-fire PIPO seedling/sapling recruitment, the data we have are all from ~15 years after the fires. So we’re using that early seral plant community that we see 15 years after fire as an indicator of the context of species and interactions the PIPO seedling experience throughout that post-fire recruitment period. The communities might be quite dynamic during that time. But at least we have this simplified window as a starting point for asking whether community matters. And we aren’t actually testing for species/community interactions directly, we’re just thinking about the patterns in the context of potential facilitation, competition, neutralism mechanisms.

(I’ll put together a full powerpoint for the entire analysis soon, and there’s still lots of work to be done to unpack the communities with a bit more details, but wanted to pass along this next phase of the work first. Hoping it might spark some feedback. )

Maybe we can have a Fall refugialites meeting...

After a deep dive into hierarchical additive models, considering the data we have, and the hypotheses we’re interested in testing, here’s the model that I’d like to work from. This is the “best” assessed by AIC and interpretability:

This is the “GI” model form from Pederson et al (2019) meaning there’s a single common smoother plus group-level smoother with different wiggliness. Information about the relationship with CMI is shared among the community types via the global smoother (but with a random term for ~intercept). Each community has a random effect for the smoother and wiggliness of the relationship akin to random effect for slope and functional form.

gam(pinpon_tally~s(pinpon.dist.600, bs="tp") + s(CMI, bs="tp",m=2) + s(CMI,by=clust.7,bs="tp",m=1)+s(clust.7,bs="re", k=7) + s(mean.x) + s(elevation), data=fulldata.noBC, method="REML", family=nb())

pinpon_tally: This is the tally of PIPO recruits within the 100-m2 plots. Includes the full bulk of zeroes, and using NB model structure

pinpon.dist.600: This is the field-measured distance from plot to nearest PIPO fire refugium. Any value labelled as 600 represents plots where no PIPO refugium was detectable. Fit as a simple s() function that shows a predictable decrease in pinpon_tally with distance from PIPO fire refugium (Figure 1).

clust.7: These are the seven community types developed using community data translated as growth forms and phenology. The clusters are generally: 

•	type 1. Bit of everything, but higher perennials. (n=158). A bit of everything, but higher perennial graminoids and sometimes higher perennial forbs
•	type 2. Annual forbs and grasses. (n=14). Higher values of annual forbs and annual graminoids
•	type 3. Evergreen shrubs. (n=39). Higher values of evergreen shrubs, moderate levels of perennial graminoids
•	type 4. Deciduous shrubs. (n=34). Higher values of deciduous shrubs, moderate levels of perennial graminoids
•	type 5. Evergreen trees. (n=51). Higher values of evergreen trees, some higher deciduous trees, perennial graminoids
•	type 6. Deciduous trees. (n=39). Higher values deciduous trees
•	type 7. Bit of everything, more woody species (n=184). A bit of everything, but higher deciduous trees, evergreen trees, evergreen shrubs than type 1 (type 1 and type 7 were the outcomes of the second round of clustering), lowBer amoungs of perennial graminoids and forbs.

CMI: This is the 1991-2020 climatological norms for Hogg’s Climate Moisture Index, and indicator of drought calculated as difference between annual precipitation and potential evapotranspiration. Positive (higher) CMI indicate wet or moist conditions, and negative (lower) values are drier. Of all the synthetic water balance variables that Sandra provided (from ClimateNA), the CMI provided the highest explanatory power. Other candidates included CMD (Hargreave’s climatic moisture index), SHM (Summer heat moisture index), the post-fire CWD percentiles (e.g., q90.x, q75.x, mean.x), and the PC1/PC2 ordinations. The goal here was to identify the best “normal” water balance metric that represents spatial gradient of the growing conditions of the plots. 

In model testing, the CMI had most explanatory power using the GI framework that fits a global shared curve, and then allowing for different functional forms and wiggliness using the random effects (Figure 1). For our purposes, we’re interested in the fact that there’s a) significant support for including the community type random effects (as s(clust.7), via AIC and p-values), b) a significant global relationship with CMI (as s(CMI)), c) significant unique relationships with CMI within community types (e.g., s(clust.7:CMI.71), all after accounting for pinpon.dist.600 (distance to seed source).

Model exploration to test for additional influence of terrain variables demonstrated no model improvement with heat.load, topo.position, etc. 

mean.x: This is a metric Sandra developed to quantify the percentiles of post-fire Climatic Water Deficit (CWD). This variable strengthened the model and had an intuitive negative linear form where as meanCWD in years post-fire increased, PIPO recruitment decreased (Figure 1). The mean.x was strongest of max.x, q75.x. The q90.x was strongest overall but looked overfit with uninterpretable wiggly shape. 

I removed the Burnt Cabin sites from the dataset after exploratory analyses because they were anomalously high in their CMI values – their own population when taken across this gradient – and only occurred in community type 4 (clust.7 = 4, deciduous shrubs) . We can revisit that later if/as needed.

Note, to goal wasn’t to develop the best overall predictive model, so we don’t continue adding in more variables. The goal is to look for a strong (set of) model(s) that explain variability in PIPO, and see whether relationships with the key water balance variable differ among community types. 

**### EMERGING STORY LINE:** 
1) We see that according to our models and data, community matters in this ecosystem, though it is secondary to major factors like seed source (pinpon.dist.600) and the overall hump-shaped climate niche of PIPO (CMI). 
Caveat: Community type is strongest is cluster/types with greatest sample size, so there might be a sample effect here – (should I do iterative set of sub-samples to balance sample numbers a bit more?)
2) There are two ways of looking for evidence that community matters, to complement the tabular output of the model; and bringing those two ways together provides a holistic assessment of the community effect. First way is to look at Figure 1 and interpret the significance and functional forms of the responses among community types (clust.7,), where and how are there differences among community types, and what might they mean? The second is to look at Figure 2 and identify differences in the PIPO tally numbers overall, and differences in the peaks/positions of the predictions across the range of CMI – again where, how, why.
•	Some community types have very low PIPO recruitment (community 2 and 4), even after accounting for distance to refugium, elevation, post-fire climate, and overall CMI. Lower levels of recruitment for a given CMI at levels of CMI where other community types have more robust recruitment. Evidence for community effect? 
•	High values of PIPO recruitment extend to broader ranges of CMI in community type 7, when e.g., compared against community type 1. Evidence for community effect? 
•	For community 5 we see higher values of PIPO recruitment overall for a given CMI in comparison to other community types.
 
**### RESULTS** 

Family: Negative Binomial(0.474) 
Link function: log 

pinpon_tally ~ s(pinpon.dist.600, bs = "tp") + s(CMI, bs = "tp", m = 2) + s(CMI, by = clust.7, bs = "tp", m = 1) + s(clust.7, bs = "re", k = 7) + s(mean.x) + s(elevation)

Parametric coefficients:
            Estimate Std. Error z value Pr(>|z|)  
(Intercept)   0.7271     0.3540   2.054     0.04 *

Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Approximate significance of smooth terms:
                         edf Ref.df  Chi.sq  p-value    
s(pinpon.dist.600) 2.774e+00  3.369 110.609  < 2e-16 ***
s(CMI)             4.384e+00  5.271  52.232  < 2e-16 ***
s(CMI):clust.71    5.038e+00  8.000  20.218 0.001254 ** 
s(CMI):clust.72    4.806e-05  8.000   0.000 0.957678    
s(CMI):clust.73    1.788e-04  8.000   0.000 0.558127    
s(CMI):clust.74    8.818e-01  8.000   2.620 0.116551    
s(CMI):clust.75    8.467e-04  8.000   0.001 0.551817    
s(CMI):clust.76    4.613e-01  8.000   0.697 0.248496    
s(CMI):clust.77    3.724e+00  8.000  22.573 0.000203 ***
s(clust.7)         5.354e+00  6.000  60.996  < 2e-16 ***
s(mean.x)          1.000e+00  1.001   9.590 0.001959 ** 
s(elevation)       1.000e+00  1.001  10.035 0.001540 ** 

Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

R-sq.(adj) =  0.202   Deviance explained = 50.1%
-REML = 1113.8  Scale est. = 1         n = 518

AIC = 2210.7

![image](https://user-images.githubusercontent.com/84547345/191135412-1573fcd9-b462-444a-89c1-1940c89ece22.png)

Figure 1. Partial effects plots showing the shapes of relationships fitted in the model. (model naming cm7.hGI.noBC.d). See Figure 2 with more descriptive labels

 
![image](https://user-images.githubusercontent.com/84547345/191135437-f3ecee33-78c0-448e-b38c-8fd2e0993566.png)

Figure 2. Predicted PIPO tally for the model, holding pinpon.dist600 at median (39.4), elevation at median (2322.1m) and mean.x at median (50.91). Colored points show the observed value of PIPO tally from the data. Note, y-axis is truncated to c(0,75) and for community 3 there is one single point at PIPO tally =180, and for community 5 there are two points at PIPO tally = 85 that push the humps in the fitted estimates and standard errors up but you don’t see here.



**### Next steps:**
1.	look at dominant species in each community type to interpret the communities, calculate species diversity metrics for sites and community types
2.	develop some more plots to support analyses (predictions, boxplots of variables, maps)
3.	input from the group – does this make sense, should I go a different direction?
4.	start fleshing out text – thinking maybe a first try at Ecology Letters?
